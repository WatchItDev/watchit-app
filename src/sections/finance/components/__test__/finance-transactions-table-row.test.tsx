import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import FinanceTransactionTableRow from "../finance-transactions-table-row";

describe("[COMPONENTS] <FinanceTransactionTableRow/>", () => {
  it("to match snapshot", () => {
    const { container } = render(
      <FinanceTransactionTableRow
        row={{
          date: BigInt("1234567890"),
          name: "Test User",
          amount: "100",
          avatarUrl: "https://example.com/avatar.jpg",
          message: "Test message",
          category: "income",
          id: "12345",
          type: "credit",
          status: "completed",
        }}
        selected={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
