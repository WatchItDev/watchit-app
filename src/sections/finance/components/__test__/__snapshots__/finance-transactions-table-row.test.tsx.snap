// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS] <FinanceTransactionTableRow/> > to match snapshot 1`] = `
<div>
  <tr
    class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
  >
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-8ud3sa-MuiTableCell-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circular css-wke4e9-MuiAvatar-root"
      >
        <img
          alt="TEST USER"
          class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
          src="https://example.com/avatar.jpg"
        />
      </div>
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-6sj2ec-MuiTypography-root"
        >
          Test message
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-10h1cz6-MuiTypography-root"
        >
          Test User
        </span>
      </div>
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-6sj2ec-MuiTypography-root"
        >
          13/02/2009
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-rxsw71-MuiTypography-root"
        >
          5:31 PM
        </span>
      </div>
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 css-yb11s7-MuiTypography-root"
      >
         
        100
         MMC
      </p>
    </td>
    <td
      class="MuiTableCell-root MuiTableCell-sizeMedium css-bsph2u-MuiTableCell-root"
    >
      <div
        class="MuiStack-root css-1ug68w3-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-eu9vn6-MuiTypography-root"
        >
          12345
        </p>
        <button
          aria-label="Open in new window"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-n2gp7e-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <span />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </td>
  </tr>
</div>
`;
